js/components/Modal.tsx

```diff
@@ -144,12 +146,16 @@ export interface ModalHeaderProps {
  * @param title Modal title
  * @returns ModalHeader component
  */
-export const ModalHeader = ({ onClose, title }: ModalHeaderProps) => {
+export const ModalHeader = ({ onClose, title, icon }: ModalHeaderProps) => {
     return (
-        <div style={modalHeaderStyle}>
-            <span style={modalTitleStyle}>{title}</span>
+        <header style={modalHeaderStyle}>
+            <span style={modalTitleStyle}>
+                {icon && <i className={`fas fa-${icon}`} />}
+                {' '}
```

Comment:
Why not just `&nbsp;`?

Reply:
This way the whitespace will collapse if there's no icon.